internal/filesystem.go

```diff
@@ -72,9 +72,40 @@ func SystemDListener() (net.Listener, error) {
 	return net.FileListener(file)
 }
 
+type runDirListener struct {
+	listener   net.Listener
+	dirCreated bool
+}
+
+func (rdl *runDirListener) Accept() (net.Conn, error) {
+	return rdl.listener.Accept()
+}
+
+func (rdl *runDirListener) Close() error {
+	var removeErr error
+	if rdl.dirCreated {
+		removeErr = os.RemoveAll(RunDir)
+	}
+	return errors.Join(removeErr, rdl.listener.Close())
+}
+
+func (rdl *runDirListener) Addr() net.Addr {
+	return rdl.listener.Addr()
+}
+
 // ManualListener returns manually created listener with provided permissions
 func ManualListener(socket string, perm fs.FileMode) func() (net.Listener, error) {
 	return func() (net.Listener, error) {
+		dirCreated := true
+		if err := os.MkdirAll(
+			RunDir, PermUserRWGroupRW,
+		); err != nil && !errors.Is(err, os.ErrExist) {
+			if errors.Is(err, os.ErrExist) {
```

Comment:
This will never be true, because it can enter into the if, only if error is not `ErrExitst`. Because of this if the error is that the folder already exists, then the function will return error.